examples/plot_synthetic.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+==========================
+Synthetic dataset example with visualization
+==========================
+"""
+
+# %%
+from fairlearn.datasets import make_synthetic_dataset
+from fairlearn.metrics import (
+    MetricFrame,
+    false_positive_rate,
+    true_positive_rate,
+    selection_rate,
+    count
+)
+import numpy as np
+from sklearn.metrics import accuracy_score, precision_score, recall_score
+from sklearn.model_selection import train_test_split
+from sklearn.tree import DecisionTreeClassifier
+
+
+rng = np.random.RandomState(seed=42)
+X, y, gender = make_synthetic_dataset(rng=rng)
+
+X_train, X_test, y_train, y_test, gender_train, gender_test = train_test_split(
+    X, y, gender, test_size=0.3, random_state=rng
+)
+
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+classifier.fit(X, y)
+y_pred = classifier.predict(X)
+
+# Analyze metrics using MetricFrame
+metrics = {
+    'accuracy': accuracy_score,
+    'precision': precision_score,
+    'recall': recall_score,
+    'false positive rate': false_positive_rate,
+    'true positive rate': true_positive_rate,
+    'selection rate': selection_rate,
+    'count': count}
+metric_frame = MetricFrame(metrics=metrics,
+                           y_true=y,
+                           y_pred=y_pred,
+                           sensitive_features=gender)
+metric_frame.by_group.plot.bar(
+    subplots=True,
+    layout=[3, 3],
+    legend=False,
+    figsize=[12, 8],
+    title="Show all metrics",
+)
+
+# Customize plots with kind
```

Comment:
I think the bar plot is sufficient. Was there anything in particular you want to show with the pie?

Reply:
Not really. It was just easy enough to include. I'll remove it.